pkg/reconciler/namespace/namespace.go

```diff
@@ -100,55 +103,156 @@ func (c *Controller) reconcileGVR(ctx context.Context, gvr schema.GroupVersionRe
 	return nil
 }
 
-// reconcileNamespace is responsible for assigning a namespace to a cluster, if
-// it does not already have one.
-//
-// After assigning (or if it's already assigned), this also updates all
-// resources in the namespace to be assigned to the namespace's cluster.
-func (c *Controller) reconcileNamespace(ctx context.Context, lclusterName string, ns *corev1.Namespace) error {
-	klog.Infof("Reconciling Namespace %s", ns.Name)
+func (c *Controller) patchStatus(ctx context.Context, ns *corev1.Namespace, updateConditions updateConditionsFunc) error {
+	patchBytes, err := statusPatchBytes(ns, updateConditions)
+	if err != nil {
+		return err
+	}
+	_, err = c.namespaceClient.Patch(ctx, ns.Name, types.MergePatchType, patchBytes, metav1.PatchOptions{}, "status")
+	if err != nil {
+		return fmt.Errorf("failed to patch status on namespace %q: %w", ns.Name, err)
```

Comment:
I'm guessing all errors and logging should be qualifying resource names with the org+workspace?